### DISCUSSION TITLE
Gene Products URL error

### DISCUSSION THREAD

**Discussion by Anisah-bb** [wikipathways/wikipathways-help#126] [General]
I ran a query to get geneProducts for a pathway. However, the URLs are not resolving. I get an error 404.  

![Image 09-08-2024 at 14 07](https://github.com/user-attachments/assets/5a770fa7-68c5-4e9c-952a-3c50ddbfb244)
![Image 09-08-2024 at 14 07](https://github.com/user-attachments/assets/7e5a6065-19b2-40b3-bcaf-97f3d1cb1ba7)

**Comment by khanspers** [comment]
Hi @Anisah-bb, sorry for the late reply. I believe this problem is caused by the rdf referencing a revision of the pathway that is not the latest revision on the website. @marvinm2 @egonw : Can you take a look at what is required to fix this?

**Comment by egonw** [comment]
@khanspers, @Anisah-bb, we have discussed this in the past, but it's not easy to track all pathway revision numbers, which is specific for MediaWiki and doesn't really translate to GitHub (cc @AlexanderPico, @mkutmon).

What I will propose to @DeniseSl22 and @marvinm2 is to update the SPARQL queries to use the revision-less pathway URL, like this:

```sparql
prefix dc: <http://purl.org/dc/elements/1.1/>

select distinct ?pathway (str(?label) as ?geneProduct) where {
    ?geneProduct a wp:GeneProduct . 
    ?geneProduct rdfs:label ?label .
    ?geneProduct dcterms:isPartOf ?pathwayRev .
    ?pathwayRev a wp:Pathway .
    ?pathwayRev dc:identifier ?pathway .
    ?pathwayRev dcterms:identifier "WP1560" . #Replace "WP1560" with WP ID of interest
}
```

Update: Proposed change: https://github.com/wikipathways/SPARQLQueries/pull/22

**Reply by Chris-Evelo** [reply]
Shouldn't the pathwayRev variable then just be called "pathway". Or did the "Rev" part not indicate that it was about a specific revision?

**Reply by egonw** [reply]
`?pathwayRev` will have the URL with revision information and `?pathway` without the revision.
